serverless-plugin/config-schema.js

```diff
@@ -0,0 +1,177 @@
+  },
+  ComparisonOperator: {
+    type: ['string', 'null'],
+    enum: [
+      'GreaterThanOrEqualToThreshold', 'GreaterThanThreshold', 'GreaterThanUpperThreshold', 'LessThanLowerOrGreaterThanUpperThreshold', 'LessThanLowerThreshold', 'LessThanOrEqualToThreshold', 'LessThanThreshold'
+    ]
+  },
+  EvaluationPeriods: {
+    type: ['integer', 'null'],
+    minimum: 1
+  },
+  Statistic: statisticType,
+  ExtendedStatistic: {
+    type: ['string', 'null'],
+    pattern: percentilePattern
+  }
+}
+
+const alarmSchemas = {}
+for (const service of Object.keys(supportedAlarms)) {
+  alarmSchemas[service] = {
+    properties: {
+      ...commonAlarmProperties
+    },
+    additionalProperties: false
+  }
+  for (const metricAlarm of supportedAlarms[service]) {
+    alarmSchemas[service].properties[metricAlarm] = {
+      properties: {
+        ...commonAlarmProperties
+      },
+      additionalProperties: false
+    }
+  }
+}
+
+const alarmsSchema = {
+  type: 'object',
+  properties: {
+    ...commonAlarmProperties,
+    ...alarmSchemas
+  },
+  additionalProperties: false
+}
+
+const commonWidgetProperties = {
+  width: { type: ['integer', 'null'], minimum: 1, maximum: 24 },
+  height: { type: ['integer', 'null'], minimum: 1, maximum: 1000 },
+  metricPeriod: { type: ['integer', 'null'], minimum: 60, multipleOf: 60 },
+  Statistic: {
+    type: 'array',
+    items: {
+      oneOf: [
+        statisticType,
+        {
+          type: 'string',
+          pattern: percentilePattern
+        }
+      ]
+    }
+  }
+}
+
+const widgetSchemas = {}
+for (const service of Object.keys(supportedWidgets)) {
+  widgetSchemas[service] = {
+    properties: {
+      ...commonWidgetProperties
+    },
+    additionalProperties: false
+  }
+  for (const metricWidget of supportedWidgets[service]) {
+    widgetSchemas[service].properties[metricWidget] = {
+      properties: {
+        ...commonWidgetProperties
+      },
+      additionalProperties: false
+    }
+  }
+}
+
+const dashboardSchema = {
+  type: 'object',
+  properties: {
+    timeRange: {
+      type: 'object',
+      oneOf: [{
+        properties: {
+          start: {
+            type: ['string', 'null'],
+            pattern: '^-P(T\\d+[MH]|\\d+[DWM]$)'
+          }
+        }
+      },
+      {
+        properties: {
+          start: { type: ['string', 'null'], pattern: iso8601Pattern },
+          end: { type: ['string', 'null'], pattern: iso8601Pattern }
+        }
+      }]
+    },
+    widgets: {
+      properties: {
+        ...commonWidgetProperties,
+        ...widgetSchemas
+      }
+    }
+  },
+  additionalProperties: false
+}
+
+/**
+ * JSON Schema for SLIC Watch configuration
+ */
+const schema = {
+  $schema: 'http://json-schema.org/draft-07/schema',
+  title: 'SLIC Watch Serverless Plugin configuration',
+  type: 'object',
+  properties: {
+    alarms: alarmsSchema,
+    dashboard: dashboardSchema,
+    topicArn: { type: ['string', 'null'] }
+  },
+  required: ['topicArn'],
```

Comment:
If this one is required, do we need to support `null` as a type?

Reply:
Nice catch, it should not support null!